File: Leetcode/sbgo/Leetcode383-赎金信.go
package sbgo

func canConstruct(ransomNote string, magazine string) bool {
	if len(magazine) < len(ransomNote) {
		return false
	}
	//'a'和ch都是int32
	words := [26]int{}
	for _, ch := range magazine {
		words[ch-'a']++
	}
	for _, ch := range ransomNote {
		words[ch-'a']--
		if words[ch-'a'] < 0 {
			return false
		}
	}
	return true
}

File: Leetcode/JS/Leetcode383-赎金信.js
var canConstruct = function(ransomNote, magazine) {
    let mp = new Array(26).fill(0);
    for(const ch of magazine){
        mp[ch.charCodeAt(0) - 'a'.charCodeAt(0)]++;
    }
    for(const ch of ransomNote) {
        mp[ch.charCodeAt(0) - 'a'.charCodeAt(0)]--;
        if(mp[ch.charCodeAt(0) - 'a'.charCodeAt(0)] < 0) return false;
    }
    return true;
};